controllers/users/checkAllowAnonDM.js

```diff
@@ -0,0 +1,21 @@
+const {User} = require('../../databases/models');
+
+module.exports = async (req, res) => {
+  let {user_id} = req.params;
+  let user = await User.findOne({where: {user_id, is_anonymous: false}});
+
+  if (user) {
+    return res.status(200).json({
+      status: 'success',
+      user: {
+        id: user.user_id,
+        allow_anon_dm: user.allow_anon_dm
+      }
+    });
+  } else {
+    return res.status(400).json({
+      status: 'error',
+      message: 'User Not Found'
+    });
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic of the code seems fine, but there's a potential security issue. You're directly exposing user IDs to the client side which can be a sensitive piece of information. It would be better to not include the user ID in the response.
```diff
  if (user) {
    return res.status(200).json({
      status: 'success',
      user: {
-        id: user.user_id,
+        // id: user.user_id, // remove this line
        allow_anon_dm: user.allow_anon_dm
      }
    });
  } else {
    return res.status(400).json({
      status: 'error',
      message: 'User Not Found'
    });
  }
};
```
Also, consider adding error handling for the database query. If the `findOne` operation fails due to a database error, it will throw an exception that is not currently being caught, which could lead to unhandled promise rejection.

```diff
module.exports = async (req, res) => {
  let {user_id} = req.params;
- 5:   let user = await User.findOne({where: {user_id, is_anonymous: false}});
+ 5:   let user;
+ 6:   try {
+ 7:     user = await User.findOne({where: {user_id, is_anonymous: false}});
+ 8:   } catch (err) {
+ 9:     console.error(err);
+ 10:    return res.status(500).json({
+ 11:      status: 'error',
+ 12:      message: 'Database Error'
+ 13:    });
+ 14:  }
...
```
This way, you can provide a more meaningful error message to the client and avoid potential server crashes due to unhandled exceptions.


<!-- This is an auto-generated comment by OSS CodeRabbit -->